util.ts

```diff
@@ -1,5 +1,7 @@
 import * as T from "./types.ts";
 
+export { guardAll, typeFromProps } from "./utils/propertiesMatcher.ts";
```

Comment:
Here are the two main functions that I created, don't have to be used. 
typeFromProps is the ConfigSpec into matcher